docs/source/crawler/basics.rst

```diff
@@ -27,5 +27,22 @@ a modified dictionary as its output once.
    or modified.  To help with creation of a new shallow copy with some fields
    adjusted, use :func:`~datalad.utils.updated`.
 
-Pipeline
---------
\ No newline at end of file
+Pipelines
+---------
+
+A pipeline is a series of functions gathered into a list. Each function take the
+output of its predecessor as its own input. The first function in the pipeline
+would need to be provided with input. The simplest pipeline could look like
+
+>>> def pipeline():
+...     [
+...     crawl_url('http://map.org/datasets'),
+...     a_href_match(".*.mat"),
+...     annex
+...     ]
+
+which crawls in a website, which is provided as input to the first function, then
+matches to all files that end in `.mat`, and those files are lastly inputted to
```

Comment:
note that it is a regular expression, so ".*.mat" would also match "blahmat" -- you need to have ".*\.mat"